curvefs/src/client/curve_fuse_op.cpp

```diff
@@ -265,17 +268,15 @@ void FuseOpGetXattr(fuse_req_t req, fuse_ino_t ino, const char *name,
     LatencyUpdater updater(&g_clientOpMetric->opGetXattr.latency);
     std::unique_ptr<char[]> buf(new char[MAXXATTRLENGTH]);
     std::memset(buf.get(), 0, MAXXATTRLENGTH);
-    CURVEFS_ERROR ret = g_ClientInstance->FuseOpGetXattr(req, ino, name,
-                                                         buf.get(), size);
-    if (ret != CURVEFS_ERROR::OK && ret != CURVEFS_ERROR::NODATA) {
-        g_clientOpMetric->opGetXattr.ecount << 1;
-        FuseReplyErrByErrCode(req, ret);
-        return;
-    }
-
     if (size == 0) {
-        fuse_reply_xattr(req, strlen(buf.get()));
+        fuse_reply_xattr(req, MAXXATTRLENGTH);
     } else {
+        CURVEFS_ERROR ret = g_ClientInstance->FuseOpGetXattr(req, ino, name,
```

Comment:
there's no need to allocated `buf` on heap, `char buf[MAXXATTRLENGTH] = {0}` is enough.

Reply:
> there's no need to allocated `buf` on heap, `char buf[MAXXATTRLENGTH] = {0}` is enough.

fixed